rdr_service/alembic/versions/813b74f84338_retention_eligible_import_table.py

```diff
@@ -0,0 +1,92 @@
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('retention_eligible_metrics',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('created', rdr_service.model.utils.UTCDateTime6(fsp=6), nullable=True),
+    sa.Column('modified', rdr_service.model.utils.UTCDateTime6(fsp=6), nullable=True),
+    sa.Column('participant_id', sa.Integer(), nullable=False),
+    sa.Column('retention_eligible', sa.Boolean(), nullable=True),
+    sa.Column('retention_eligible_time', rdr_service.model.utils.UTCDateTime6(fsp=6), nullable=True),
+    sa.Column('actively_retained', sa.Boolean(), nullable=True),
+    sa.Column('passively_retained', sa.Boolean(), nullable=True),
+    sa.Column('file_upload_date', rdr_service.model.utils.UTCDateTime6(fsp=6), nullable=True),
+    sa.Column('retention_eligible_status', rdr_service.model.utils.Enum(RetentionStatus), nullable=True),
+    sa.Column('retention_type', rdr_service.model.utils.Enum(RetentionType), nullable=True),
+    sa.ForeignKeyConstraint(['participant_id'], ['participant.participant_id'], ),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_index(op.f('ix_retention_eligible_metrics_participant_id'), 'retention_eligible_metrics', ['participant_id'], unique=True)
+
+    op.drop_column('participant_summary', 'retention_eligible_time')
+    op.drop_column('participant_summary', 'retention_eligible_status')
+
+    op.add_column('participant_summary',
+                  sa.Column('retention_eligible_time', rdr_service.model.utils.UTCDateTime6(fsp=6), nullable=True))
+    op.add_column('participant_summary',
+                  sa.Column('retention_eligible_status', rdr_service.model.utils.Enum(RetentionStatus), nullable=True))
+    op.add_column('participant_summary',
+                  sa.Column('retention_type', rdr_service.model.utils.Enum(RetentionType), nullable=True))
+    # ### end Alembic commands ###
+
+
+def downgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_column('participant_summary', 'retention_type')
+    op.drop_column('participant_summary', 'retention_eligible_time')
+    op.drop_column('participant_summary', 'retention_eligible_status')
+    op.add_column('participant_summary',
+                  sa.Column('retention_eligible_status', rdr_service.model.utils.Enum(RetentionStatus),
+                            sa.Computed(
+                                '\n    CASE WHEN\n      consent_for_study_enrollment = 1\n      AND (consent_for_electronic_health_records = 1 OR consent_for_dv_electronic_health_records_sharing = 1)\n      AND questionnaire_on_the_basics = 1\n      AND questionnaire_on_overall_health = 1\n      AND questionnaire_on_lifestyle = 1\n      AND withdrawal_status = 1\n      AND suspension_status = 1\n      AND samples_to_isolate_dna = 1\n    THEN 2 ELSE 1\n    END\n',
+                                persisted=True), nullable=True))
```

Comment:
Can these computed SQL statements be put in a multi-line Python string variable above in this file for readability, then use the variable in the 'sa.Computed()' function?

Reply:
Robert, these strings are part of the auto-generated migration and would be for adding back the computed columns if we were to downgrade. I imagine in that scenario we'd refer to the string at https://github.com/all-of-us/raw-data-repository/pull/2198/files#diff-ec950a38bbe9afabace937c2d149d8625b967edae25d41d47df7a676608e704dL74 for readability if needed and would rather leave the migration as is. How does that sound?